expo/expo
# Allow `undefined` onBarCodeScanned prop

# Why

Currently this line in the example code in the official docs produces a type error: 
```
onBarCodeScanned={code ? undefined : handleBarCodeScanned}
``` 
because the prop doesn't expect `undefined`. Yet it's a valid, documented value that enables the user to pause scanning. 

# How

This PR changes the type of the onBarCodeScanned prop to allow `undefined`

# Test Plan


1. Create a managed typescript expo project
1. `expo install expo-barcode-scanner`
1. Copy the official example from the docs (https://docs.expo.io/versions/latest/sdk/bar-code-scanner/)
1. Notice the type error as described above.
1. Pull this PR, and notice no more error.
1. yarn build successful
